VideoWeb/VideoWeb/ClientApp/src/app/on-the-day/quick-links/quick-links.component.spec.ts

```diff
@@ -0,0 +1,200 @@
+        it('should call method to initialise errors', () => {
+            const spy = spyOn(component, 'resetErrors');
+            component.ngOnInit();
+            expect(spy.calls.count()).toBe(1);
+        });
+
+        it('should call method to initialise the form', () => {
+            const spy = spyOn(component, 'initialiseForm');
+            component.ngOnInit();
+            expect(spy.calls.count()).toBe(1);
+        });
+
+        it('should call quick links service to get participant roles if quick link is valid', () => {
+            expect(quickLinksServiceSpy.getQuickLinkParticipantRoles.calls.count()).toBe(1);
+            expect(component.quickLinkParticipantRoles).toEqual(quickLinkParticipantRoles);
+        });
+    });
+
+    describe('initialiseForm', () => {
+        it('should initialise the form', () => {
+            component.initialiseForm();
+
+            expect(component.quickLinkForm.controls['name'].value).toBe('');
+            expect(component.quickLinkForm.controls['name'].valid).toBeFalse();
+            expect(component.quickLinkForm.controls['quickLinkParticipantRole'].value).toBe('');
+            expect(component.quickLinkForm.controls['quickLinkParticipantRole'].valid).toBeFalse();
+        });
+    });
+
+    describe('validateForm', () => {
+        it('should set name error and mark form as invalid if name is not populated', () => {
+            component.isFormValid = true;
+            component.error.nameError = '';
+            component.quickLinkForm.controls['name'].setValue('');
+
+            component.validateForm();
+
+            expect(component.isFormValid).toBeFalse();
+            expect(component.error.nameError).toBe('Please enter your full name');
+        });
+
+        it('should set role error and mark form as invalid if name is not populated', () => {
+            component.isFormValid = true;
+            component.error.roleError = '';
+            component.quickLinkForm.controls['quickLinkParticipantRole'].setValue('');
+
+            component.validateForm();
+
+            expect(component.isFormValid).toBeFalse();
+            expect(component.error.roleError).toBe('Please choose your role in the hearing');
+        });
+
+        it('should mark form as valid if form validations are all passed', () => {
+            component.isFormValid = false;
+            component.quickLinkForm.controls['name'].setValue('name');
+            component.quickLinkForm.controls['quickLinkParticipantRole'].setValue('quickLinkParticipantRole');
+
+            component.validateForm();
+
+            expect(component.isFormValid).toBeTrue();
+        });
+    });
+
+    describe('resetErrors', () => {
+        it('should resets errors', () => {
+            component.error.nameError = component.error.roleError = 'error';
+            component.resetErrors();
+            expect(component.error.nameError).toBe('');
+            expect(component.error.nameError).toBe('');
```

Comment:
Checks the same error. Probably should set a role error in the arrange and check that it is empty in the assert

Reply:
Removed the repetition